scripts/deploy-contracts-live.sh

```diff
@@ -154,10 +154,14 @@ deploy_contracts() {
 
   # instantiate Leaser
   local -r leaser_init_msg='{"lease_code_id":"'"$lease_code_id"'","lpp":"'"$lpp_contract_address"'","profit":"'"$profit_contract_address"'","time_alarms":"'"$timealarms_contract_address"'","market_price_oracle":"'"$oracle_contract_address"'","lease_position_spec":{"liability":{"initial":650,"healthy":700,"first_liq_warn":720,"second_liq_warn":750,"third_liq_warn":780,"max":800,"recalc_time":7200000000000},"min_asset":{"amount":"150","ticker":"'"$protocol_currency"'"},"min_transaction":{"amount":"10","ticker":"'"$protocol_currency"'"}},"lease_interest_rate_margin":30,"lease_due_period":5184000000000000,"dex":{"connection_id":"'"$dex_connection"'","transfer_channel":{"local_endpoint":"'"$dex_channel_local"'","remote_endpoint":"'"$dex_channel_remote"'"}}}'
-  local -r leaser_contract_address=$(_instantiate "$nolus_net" "$chain_id" "$nolus_home_dir" "$dex_admin_wallet_key" "$leaser_code_id" "$leaser_init_msg" "$protocol-leaser" "$protocol" "$leaser_expected_address" "$admin_contract_address" )
+  local -r leaser_contract_address=$(_instantiate "$nolus_net" "$chain_id" "$nolus_home_dir" "$dex_admin_wallet_key" "$leaser_code_id" "$leaser_init_msg" "$protocol-leaser" "$protocol" "$leaser_expected_address" "$admin_contract_address")
+
+  # upload and instantiate Reserve
+  local -r reserve_init_msg='{"lease_code_admin":"'"$leaser_contract_address"'","lease_code_id":"'"$lease_code_id"'"}'
+  local -r reserve_contract_address=$(_deploy_contract "$nolus_net" "$chain_id" "$nolus_home_dir" "$dex_admin_wallet_key" "$store_code_privileged_wallet_key" "$admin_contract_address" "$wasm_path/reserve.wasm" "$reserve_init_msg" "$protocol-reserve" "$protocol")
 
   # register the protocol
-  local -r add_protocol_set_exec_msg='{"register_protocol":{"name":"'"$protocol"'","protocol":{"network":"'"$network"'","contracts":{"leaser":"'"$leaser_contract_address"'","lpp":"'"$lpp_contract_address"'","oracle":"'"$oracle_contract_address"'","profit":"'"$profit_contract_address"'"}}}}'
+  local -r add_protocol_set_exec_msg='{"register_protocol":{"name":"'"$protocol"'","protocol":{"network":"'"$network"'","contracts":{"leaser":"'"$leaser_contract_address"'","lpp":"'"$lpp_contract_address"'","oracle":"'"$oracle_contract_address"'","profit":"'"$profit_contract_address"'","reserve":"'"$reserve_contract_address"'"}}}}'
```

Comment:
I am afraid the Admin::register_protocol is not aware (yet) of the reserve

Reply:
Yes, but it's part of the full implementation of the Reserve, so I prepared it